libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -320,6 +320,14 @@ - (void)enqueueRequest:(SFRestRequest *)request delegate:(id<SFRestDelegate>)del
     }
 }
 
+- (SFNetwork *)networkForRequest:(SFRestRequest *)request {
+    if (request.shouldRunInBackground) {
```

Comment:
@bbirman We should also handle this scenario  which  is not common but likely to happen i.e  RestClient will be used to invoke rest api(s) from a different host altogether.i.e If  request.serviceHostType is SFSDKRestServiceHostTypeCustom we shouldn't reuse the same URLSession.

Reply:
That can still use the shared session, right?